### DISCUSSION TITLE
Async remote file explorer support

### DISCUSSION THREAD

**Discussion by alannotnerd** [nvim-tree/nvim-tree.lua#2299] [Ideas]
Maybe we should take a look at [distant](https://github.com/chipsenkbeil/distant.nvim) which gives ability to browse and edit remote file system. It's really cool to bring a remote file explorer just like vscode does.

**Comment by alannotnerd** [comment]
I guess it's better to bring a VFS layer so that we can reuse the left part.

**Comment by alex-courtis** [comment]
I'm happy to look into adding such optional functionality; there have been has been discussions:
- #145
- #316
- #1807

> (Alpha stage software) This plugin is in rapid development and may break or change frequently! construction

This concerns me, however so long as it is optional and someone can champion the functionality this should be acceptable.

[Pull Requests](https://github.com/nvim-tree/nvim-tree.lua/blob/master/CONTRIBUTING.md) are always gratefully appreciated, however it seems that this functionality may better be implemented via an [Extension Plugin](https://github.com/nvim-tree/nvim-tree.lua/wiki/Extension-Plugins), with API added as required.
